Tensile/TensileCreateLib/KernelFileContext.py

```diff
@@ -0,0 +1,156 @@
+    than 1 and `kernelFiles` is not empty, it opens files based on the first kernel file name.
+    Otherwise, if `mergeFiles` or `lazyLoading` is enabled, it opens fixed-named files for appending.
+
+    Args:
+        numMergedFiles: The number of files to merge, affecting file opening behavior.
+        mergeFiles: Flag indicating if files should be merged, affecting file naming.
+        lazyLoading: Flag indicating if lazy loading is enabled, also affecting file naming.
+        outputPath: Path to the directory for creating or appending files.
+        kernelFiles: Optional list of kernel file names, used for naming if not empty.
+
+    Returns:
+        A tuple of `TextIOWrapper` objects for the kernel source and header files, or (None, None)
+        if no files need to be opened based on the conditions.
+
+    Notes:
+        - Relies on `openFilesBasedOnFirstKernel` and `openFilesWithFixedNames` for file handling.
+        - The caller is responsible for closing the returned file objects.
+    """
+    kernelSourceFile, kernelHeaderFile = None, None
+    if numMergedFiles > 1 and kernelFiles:
+        kernelSourceFile, kernelHeaderFile = _openFilesBasedOnFirstKernel(kernelFiles)
+    elif mergeFiles or lazyLoading:
+        kernelSourceFile, kernelHeaderFile = _openFilesWithFixedNames(outputPath)
+
+    return kernelSourceFile, kernelHeaderFile
+
+
+def _closeKernelFiles(
+    kernelSourceFile: Optional[TextIOWrapper], kernelHeaderFile: Optional[TextIOWrapper]
+):
+    """Closes the kernel source and header file objects if they are open.
+
+    This function checks if the provided file objects for the kernel source and header files are
+    not None (they are open) and closes them. It's a cleanup function to ensure that file resources
+    are properly released after their use.
+
+    Args:
+        kernelSourceFile: The file object for the kernel's source code, or None if it's not open.
+        kernelHeaderFile: The file object for the kernel's header code, or None if it's not open.
+    """
+    if kernelSourceFile:
+        kernelSourceFile.close()
+    if kernelHeaderFile:
+        kernelHeaderFile.close()
+
+
+def _openFilesWithFixedNames(outputPath: Path) -> Tuple[TextIOWrapper, TextIOWrapper]:
+    """Opens files for kernel source code (Kernels.cpp) and header code (Kernels.h).
+
+    Args:
+        outputPath: The directory path where the files should be opened.
+
+    Returns:
+        A tuple containing two `TextIOWrapper` objects for the source and header files respectively.
+
+    Notes:
+        - The source file is named "Kernels.cpp" and the header file is named "Kernels.h".
+        - Both files are opened in append mode with UTF-8 encoding.
+    """
+    srcFilename = Path(outputPath).resolve() / "Kernels.cpp"
+    hdrFilename = Path(outputPath).resolve() / "Kernels.h"
+    return open(srcFilename, "a", encoding="utf-8"), open(hdrFilename, "a", encoding="utf-8")
```

Comment:
Minor style suggestion:
```suggestion
    return (open(srcFilename, "a", encoding="utf-8"),  
                open(hdrFilename, "a", encoding="utf-8"))
```